src/librustc/hir/map/mod.rs

```diff
@@ -266,10 +266,11 @@ impl<'hir> Map<'hir> {
         self.tcx.definitions.def_index_to_hir_id(def_id.to_def_id().index)
     }
 
-    pub fn def_kind(&self, hir_id: HirId) -> Option<DefKind> {
-        let node = self.find(hir_id)?;
+    pub fn def_kind(&self, local_def_id: LocalDefId) -> DefKind {
+        let hir_id = self.local_def_id_to_hir_id(local_def_id);
```

Comment:
To get this PR going you could just check early if `local_def_id.index == CRATE_DEF_INDEX` and then return `DefKind::Crate` / `DefKind::Mod`.